### PR TITLE
ssl-proxy: 0.2.7 -> 0.2.7-unstable-2024-02-05

### PR DESCRIPTION
**Pull request by konst-aa**
## Description of changes
cc: @marsam

In PR #282237, marsam unpinned `ssl-proxy`'s dependency on go 1.19 by adding a patch directly into the derivation. I put up a [PR upstream](https://github.com/suyashkumar/ssl-proxy/pull/53), and it was merged! As we don't need the patch now, I removed it and updated the fetcher to grab the commit where it was applied.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by konst-aa** [comment]
I just realized that I forgot to remove the `fetchFromGithub` dependency. I will do that when I get home. Also, is it reasonable to use `fetchGit` here?

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 1]
```diff
@@ -4,18 +4,13 @@ buildGoModule rec {
   pname = "ssl-proxy";
   version = "0.2.7";
 
-  src = fetchFromGitHub {
-    owner = "suyashkumar";
-    repo = "ssl-proxy";
-    rev = "v${version}";
-    hash = "sha256-c9BLdDlkrg1z1QrO+vEAVyPtrV/nQcYlGXFmwfAOSpQ=";
+  src = builtins.fetchGit {
```
Comment (kirillrdy) [inline pkgs/tools/networking/ssl-proxy/default.nix:7]: you should keep using fetchFromGitHub, what was the reason for change ?

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 2]
```diff
@@ -4,18 +4,13 @@ buildGoModule rec {
   pname = "ssl-proxy";
   version = "0.2.7";
```
Comment (kirillrdy) [inline pkgs/tools/networking/ssl-proxy/default.nix:5]: version needs to be updated https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

Comment (konst-aa) [reply pkgs/tools/networking/ssl-proxy/default.nix]: That makes a lot of sense, TIL. Fixed!

**Comment by kirillrdy** [comment]
@konst-aa thank you for making those changes !

 now i think the only thing that is left is to update commit and PR title 

`ssl-proxy: 0.2.7 -> 0.2.7-unstable-2024-02-05`,  you can add `remove patch that has been applied upstream` as second line in commit

**Review by kirillrdy — APPROVED** [review]

**Pull request merged**